packages/pnpm-sync/src/start.ts

```diff
@@ -58,7 +65,23 @@ program
           if (lockfile === null) {
             return undefined;
           } else {
-            return lockfile;
+            const lockfilePackages: Record<string, ILockfilePackage> = {};
+            for (const packagePath in lockfile.packages) {
+              if (lockfile.packages[packagePath]) {
+                lockfilePackages[packagePath] = {
+                  dependencies: lockfile.packages[packagePath].dependencies,
+                  optionalDependencies: lockfile.packages[packagePath].optionalDependencies
```

Comment:
Save `packages[packagePath]` in a local variable to avoid repeatedly looking it up.